### PR TITLE
Update permissions on GOPATH

### PR DESCRIPTION
**Pull request by mikenairn**
Verify:

```
$ cd openshift-ci
$ docker build -t registry.svc.ci.openshift.org/integr8ly/heimdall-base-image:latest - < Dockerfile.tools
$ IMAGE_NAME=registry.svc.ci.openshift.org/integr8ly/heimdall-base-image:latest test/run
....
SUCCESS!
$ echo $?
0
```

### CODE REVIEW

**Comment by laurafitzgerald** [comment]
The logs here are making me thing that the failures are not to do with the change. So retesting.

File: openshift-ci/README.md [hunk 1]
```diff
@@ -7,10 +7,10 @@ Base image used on CI for all builds and test jobs. See [here](https://github.co
...
-$ docker build -t registry.svc.ci.openshift.org/openshift/release:intly-golang-1.13 - < Dockerfile.tools
-$ IMAGE_NAME=registry.svc.ci.openshift.org/openshift/release:intly-golang-1.13 test/run
-operator-sdk version: "v0.12.0", commit: "2445fcda834ca4b7cf0d6c38fba6317fb219b469", go version: "go1.13.5 linux/amd64"
-go version go1.13.5 linux/amd64
-hello world
+$ docker build -t registry.svc.ci.openshift.org/integr8ly/heimdall-base-image:latest - < Dockerfile.tools
+$ IMAGE_NAME=registry.svc.ci.openshift.org/integr8ly/heimdall-base-image:latest test/run
+operator-sdk version: "v0.12.0", commit: "2445fcda834ca4b7cf0d6c38fba6317fb219b469", go version: "go1.13.5 linux/amd64"                                                                                            
+go version go1.13.5 linux/amd64  
+...
```
Comment (mikenairn) [inline openshift-ci/README.md:14]: @laurafitzgerald renamed this

File: openshift-ci/test/run [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
-echo 'package main
-import \"fmt\"
-func main() {
-    fmt.Println(\"hello world\")
-}' > main.go; go run main.go"
+
+docker run --rm -u $UID -v "${WORK_DIR}/../..:/heimdall:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /heimdall && make vendor/check"
+docker run --rm -u $UID -v "${WORK_DIR}/../..:/heimdall:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /heimdall && make code/check"
+docker run --rm -u $UID -v "${WORK_DIR}/../..:/heimdall:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /heimdall && make code/compile"
+docker run --rm -u $UID -v "${WORK_DIR}/../..:/heimdall:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /heimdall && make test/unit"
```
Comment (mikenairn) [inline openshift-ci/test/run:17]: @laurafitzgerald Modifed this Docker file test file to execute the actual make targets we would be executing on CI with a on root user.

**Pull request merged**